src/prometheus_alert_rules/host_health.rules

```diff
@@ -0,0 +1,25 @@
+groups:
+- name: HostHealth
+  rules:
+  - alert: HostUp
+    expr: up < 1
+    for: 5m
+    labels:
+      severity: critical
+    annotations:
+      summary: Host '{{labels.instance}}' is down.
+      description: >-
+        Host '{{ $labels.instance }}' is down.
+          VALUE = {{ $value }}
+          LABELS = {{ $labels }}
+  - alert: HostAvailable
+    expr: absent(up)
```

Comment:
I suspect that this will be interpreted by prometheus as "whenever `up` is empty (no results)" which is not what we're looking for. That would only return true if all hosts are down.

I think we'd need to generate a specific `absent(up{instance="xyz"})` check for each host.

Reply:
I've just understood that this is solved by the prometheus scrape library. The library picks up alert rules defined in the `src/prometheus_alert_rules` directory and before sending them to Prometheus, injects the Juju topology labels. See this snippet of the library's docstring:

```
The `MetricsEndpointProvider` will read all available alert rules and also inject "filtering labels" into the alert expressions. The filtering labels ensure that alert rules are localised to the metrics provider charm's Juju topology (application, model and its UUID). Such a topology filter is essential to ensure that alert rules submitted by one provider charm generates alerts only for that same charm. When alert rules are embedded in charm, and the charm is deployed as a Juju application, the alert rules from that application have their expressions automatically updated to filter for metrics coming from the units of that application alone.
```